packages/@romejs/compiler/lint/rules/react/noUselessFragment.ts

```diff
@@ -0,0 +1,74 @@
+import {Path, TransformExitResult} from "@romejs/compiler";
+import {descriptions} from "@romejs/diagnostics";
+import {
+	doesNodeMatchPattern,
+	hasJSXAttribute,
+	isFunctionNode,
+	isJSXElement,
+} from "@romejs/js-ast-utils";
+import {AnyNode, JSXElement, JSXFragment} from "@romejs/ast";
+import {REDUCE_REMOVE} from "@romejs/compiler/constants";
+
+function isChildOfHtmlElement(path: Path): boolean {
+	const parentNode = path.parentPath.node;
+	return (
+		parentNode.type === "JSXElement" &&
+		parentNode.name.type === "JSXIdentifier" &&
+		/^[a-z]+$/.test(parentNode.name.name)
+	);
+}
+
+function isFragment(node: JSXFragment | JSXElement): boolean {
+	return (
+		node.type === "JSXFragment" ||
+		(isJSXElement(node) &&
+		(doesNodeMatchPattern(node.name, "Fragment") ||
+		doesNodeMatchPattern(node.name, "React.Fragment")))
+	);
+}
+
+function hasLessThanTwoChildren(node: JSXFragment | JSXElement): boolean {
+	return node.children.length < 2;
+}
+
+function getChildrenNode(
+	node: JSXFragment | JSXElement,
+): AnyNode | typeof REDUCE_REMOVE | Array<AnyNode> {
+	if (node.children.length === 0) {
+		return REDUCE_REMOVE;
+	}
+	if (node.children.length === 1) {
+		return node.children[0];
+	}
+	return node.children;
+}
+
+export default {
+	name: "noUselessFragment",
+	enter(path: Path): TransformExitResult {
+		const {node, context} = path;
+
+		if (node.type !== "JSXFragment" && !isJSXElement(node)) {
+			return node;
+		}
+
+		if (
+			path.parentPath.node.type !== "JSReturnStatement" &&
+			path.parentPath.node.type !== "JSVariableDeclarator" &&
```

Comment:
You should just be able to use `path.parent` here

```suggestion
			path.parent.type !== "JSReturnStatement" &&
			path.parent.node.type !== "JSVariableDeclarator" &&
```

Reply:
I can make the change. When should one be used over the other? (so I know how to think about it in the future)

Reply:
Lol path.parent just points to path.ParentPath.node. NVM